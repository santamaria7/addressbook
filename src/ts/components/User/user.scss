.user {
  display: flex;
  margin: 10px 0;
  padding: 15px;
  box-shadow: 1px 1px 8px rgba(8, 79, 90, 0.38);
  transition: all ease 0.3s;
  @media screen and (min-width: 1200px) {
    &:hover {
      box-shadow: 1px 1px 8px rgba(8, 79, 90, 0.58);
      cursor: pointer;
    }
  }
  &__img {
    margin: 0 10px 0 0;
    width: 50px;
    height: 50px;
    overflow: hidden;
    img {
      width: 100%;
      height: 100%;
      object-fit: scale-down;
      object-position: center;
    }
  }
  p {
    font-size: 14px;
    margin: 0;
    &:first-child {
      font-weight: bold;
    }
    &:nth-child(2) {
      margin: 2px 0 5px;
    }
    &:last-child {
      font-size: 13px;
      color: #333;
    }
  }
}
