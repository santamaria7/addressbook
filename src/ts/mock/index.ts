import configureMockStore from "redux-mock-store";
import createSagaMiddleware from "redux-saga";
import rootSaga from "../store/saga";
import { configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
export function mockStore(){
    configure({ adapter: new Adapter() });
    const sagaMiddleware = createSagaMiddleware();
    const mockStoreAPI = configureMockStore([sagaMiddleware]);

    const store = mockStoreAPI({
        users: [],
        pagination: {
            limit: 1000,
            offset: 1,
        },
        search: {
            isFiltered: false,
        },
        loading: false,
        nationality: "",
    });
    sagaMiddleware.run(rootSaga);
    return store
}