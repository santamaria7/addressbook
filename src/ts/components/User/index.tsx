import React from "react";
import "./user.scss";

const UserItem: React.FC<UserProps> = ({ user }) => {
  const showDetails = () => {
    // TODO: open details modal
    console.log(user);
  };

  return (
    <div className="user" onClick={showDetails}>
      <div className="user__img">
        <img
          src={user.picture.thumbnail}
          alt={`${user.name.first} ${user.name.last}`}
        />
      </div>
      <span className="user__name">{`${user.name.first} ${user.name.last}`}</span>
    </div>
  );
};

export default UserItem;
