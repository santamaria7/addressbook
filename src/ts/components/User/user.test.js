import React from "react";
import renderer from "react-test-renderer";
import User from "./index";
import { mockUser } from "../../../../mock";
const { act } = renderer;

describe("render user component", function () {
  it("should render user component with mock data", function () {
    const component = renderer.create(<User user={mockUser()} />);
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
    act(() => {
      tree.props.onClick();
      // re-rendering
      tree = component.toJSON();
      expect(tree).toMatchSnapshot();
    });
  });
});
