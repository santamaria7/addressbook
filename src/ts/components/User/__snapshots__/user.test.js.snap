// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render user component should render user component with mock data 1`] = `
<div
  className="user"
  onClick={[Function]}
>
  <div
    className="user__img"
  >
    <img
      alt="brad gibson"
      src="https://randomuser.me/api/portraits/thumb/men/75.jpg"
    />
  </div>
  <div
    className="user__details"
  >
    <p>
      silverswan131
    </p>
    <p>
      brad gibson
    </p>
    <p>
      brad.gibson@example.com
    </p>
  </div>
</div>
`;

exports[`render user component should render user component with mock data 2`] = `
<div
  className="user"
  onClick={[Function]}
>
  <div
    className="user__img"
  >
    <img
      alt="brad gibson"
      src="https://randomuser.me/api/portraits/thumb/men/75.jpg"
    />
  </div>
  <div
    className="user__details"
  >
    <p>
      silverswan131
    </p>
    <p>
      brad gibson
    </p>
    <p>
      brad.gibson@example.com
    </p>
  </div>
</div>
`;
